contrib/meson/meson/tests/meson.build

```diff
@@ -10,43 +10,81 @@
 
 lz4_source_root = '../../../..'
 
+fuzzer_time = 90
 exes = {
-  'fullbench': {
-    'sources': files(lz4_source_root / 'tests/fullbench.c'),
+  'abiTest': {
+    'sources': files(lz4_source_root / 'tests/abiTest.c'),
+    'test': false,
+  },
+  'checkFrame': {
+    'sources': files(lz4_source_root / 'tests/checkFrame.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
   },
-  'fuzzer': {
-    'sources': files(lz4_source_root / 'tests/fuzzer.c'),
+  'checkTag': {
+    'sources': files(lz4_source_root / 'tests/checkTag.c'),
+    'test': false,
+  },
+  'datagen': {
+    'sources': files(lz4_source_root / 'tests/datagencli.c'),
+    'objects': lz4.extract_objects(lz4_source_root / 'programs/datagen.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
   },
+  'decompress-partial-usingDict.c': {
+    'sources': files(lz4_source_root / 'tests/decompress-partial-usingDict.c'),
+  },
+  'decompress-partial.c': {
+    'sources': files(lz4_source_root / 'tests/decompress-partial.c'),
+  },
   'frametest': {
     'sources': files(lz4_source_root / 'tests/frametest.c'),
     'include_directories': include_directories(lz4_source_root / 'programs'),
+    'args': ['-v', '-T@0@s'.format(fuzzer_time)],
+    'test': false,
   },
-  'roundTripTest': {
-    'sources': files(lz4_source_root / 'tests/roundTripTest.c'),
+  'freestanding': {
+    'sources': files(lz4_source_root / 'tests/freestanding.c'),
+    'c_args': ['-ffreestanding', '-Wno-unused-parameter', '-Wno-declaration-after-statement'],
+    'link_args': ['-nostdlib'],
+    'test': cc.get_id() in ['gcc', 'clang'] and get_option('optimization') not in ['2', '3']
```

Comment:
Again, this validation for the cc id is needed to detect whether you can build the executable at all... but "test" only determines whether after building it, it is defined as a test.

This either needs to be a "build" key, or pull it out of the dict and check the cc id, then append to `exes +=`

Reply:
hmm yea, didn't notice this. Will fix tonight.

Reply:
Would also be cool to get MSVC + meson testing here, athough the WrapDB will eventually test it after release.

Reply:
what is the msvc equivalent of the c and link args here?

There is no msvc testing in the CI currently, unfortunately.